### PR TITLE
Update pseudorandom number generator for vpPoseRansac

### PR DESCRIPTION
**Pull request by s-trinh**
Hopefully it will fix issue with sporadic failure on ARM CI.

### CODE REVIEW

File: modules/core/src/math/random-generator/vpGaussRand.cpp [hunk 1]
```diff
@@ -48,25 +48,22 @@
 */
 double vpGaussRand::gaussianDraw()
 {
-  static bool AlreadyDone = false;
-  static double x2;
```
Comment (s-trinh) [inline modules/core/src/math/random-generator/vpGaussRand.cpp]: These static variables store a global state that prevent to have repeatable results.
```

File: modules/core/include/visp3/core/vpGaussRand.h [hunk 2]
```diff
@@ -117,18 +117,22 @@ noise 8: 10.4238
 noise 9: 10.2391
   \endcode
  */
-class VISP_EXPORT vpGaussRand : public vpUniRand
```
Comment (s-trinh) [inline modules/core/include/visp3/core/vpGaussRand.h]: No need to inherit from `vpUniRand `.

File: modules/core/src/math/random-generator/vpUniRand.cpp [hunk 3]
```diff
@@ -29,69 +29,186 @@
...
 */
-double vpUniRand::draw1()
+uint32_t vpUniRand::boundedRand(uint32_t bound)
 {
-  const long ntab = 33; // we work on a 32 elements array.
-                        // the 33rd one is actually the first value of y.
-  const long modulo = ntab - 2;
-
-  static long y = 0;
-  static long T[ntab];
```
Comment (s-trinh) [inline modules/core/src/math/random-generator/vpUniRand.cpp]: These static variables store a global state that prevent to have repeatable results.

File: modules/core/test/math/testRand.cpp [hunk 4]
```diff
@@ -0,0 +1,382 @@
...
+    chrono.stop();
+
+    double pi_error = pi - M_PI;
+    std::cout << "C++ rand() calculated pi: " << pi << " in " << chrono.getDurationMs() << " ms" << std::endl;
+    std::cout << "pi error: " << pi_error << std::endl;
+
+    CHECK(pi == Approx(M_PI).margin(0.005));
+  }
+
+  SECTION("Olf ViSP vpUniRand implementation")
```
Comment (fspindle) [inline modules/core/test/math/testRand.cpp]: Old

**Comment by fspindle** [comment]
@ s-trinh I let you merge. Ok for me.

**Pull request merged**